cpp/src/groupby/hash/groupby.cu

```diff
@@ -390,7 +390,8 @@ void sparse_to_dense_results(table_view const& keys,
                              rmm::cuda_stream_view stream,
                              rmm::mr::device_memory_resource* mr)
 {
-  auto row_bitmask{bitmask_and(keys, stream, rmm::mr::get_current_device_resource())};
+  auto row_bitmask =
+    std::move(bitmask_and(keys, stream, rmm::mr::get_current_device_resource()).mask);
```

Comment:
Can you help me understand why `std::move` is used here? In my understanding the resulting struct from `bitmask_and(...)` would be an rvalue, and accessing its member `mask` should already be an rvalue without the need for `std::move`. Is this an additional hint to the compiler or is it necessary for the desired move semantics? (This question affects several places in this PR, particularly those that don't bind the resulting struct to a name that makes it an lvalue.)

Reply:
It is redundant. Thanks for the correction. I was using `std::move` as a lazy way to silent all potential compiler errors about the deleted `rmm::device_buffer` copy ctor